### PR TITLE
feat: add get-comment-by-id endpoint and testing

### PR DESCRIPTION
**Pull request by Oseni03**
## Description

This pull request adds a new endpoint to the FastAPI application that allows users to retrieve a comment by its `comment_id`. The endpoint performs a database query to fetch the comment details and returns a JSON response with the comment information if found, or a 404 error if the comment does not exist.

## Motivation and Context

This change is required to provide an API endpoint for retrieving individual comments based on their unique identifier. It solves the problem of needing to fetch specific comment details, which is essential for various client-side functionalities such as displaying comment details in user interfaces.

## How Has This Been Tested?

The changes have been tested using `pytest` with the following tests:
- `test_get_comment_not_found`: Ensures that the endpoint returns a 404 error when a non-existent `comment_id` is requested.

Testing environment:
- SQLite database used for both development and testing.
- FastAPI's `TestClient` for endpoint testing.

## Screenshots (if appropriate - Postman, etc):

N/A

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/schemas/comment.py
Comment (joboy-dev) [inline api/v1/schemas/comment.py:1]: Anything id should be a string, not UUID

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make use of services. Check api/v1/services folder for how to do this because database queries should not be in the routes.

**Review by Goketech — APPROVED** [review]

**Comment by Goketech** [comment]
@joboy-dev

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Resolve your merge conflict

**Review by Goketech — APPROVED** [review]

**Pull request closed without merging**